src/silo/common/fasta_reader.cpp

```diff
@@ -0,0 +1,40 @@
+#include "silo/common/fasta_reader.h"
+
+#include "silo/common/fasta_format_exception.h"
+
+silo::FastaReader::FastaReader(const std::string& in_file_name)
+    : in_file(in_file_name) {}
+
+bool silo::FastaReader::nextKey(std::string& key) {
+   std::string key_with_prefix;
+   if (!getline(in_file.getInputStream(), key_with_prefix)) {
+      return false;
+   }
+
+   if (key_with_prefix.at(0) != '>') {
+      throw FastaFormatException("Fasta key prefix '>' missing for key: " + key_with_prefix);
+   }
+
+   key = key_with_prefix.substr(1);
+
+   in_file.getInputStream().ignore(LONG_MAX, '\n');
+   return true;
+}
+
+bool silo::FastaReader::next(std::string& key, std::string& genome) {
+   nextKey(key);
```

Comment:
I think it's unfortunately not that easy. Don't we need a `return false` somehow?

Reply:
Hmm, after consideration I would leave this change out. The repetition is minimal and we would need to mutate the key variable earlier, which is not wanted in case we throw an exception and the nextKey function would need to be passed an additional parameter, to consider the case whether the genome is skipped or not.

Reply:
How about that?
```
bool populateKey(std::string& key) {
   std::string key_with_prefix;
   if (!getline(in_file.getInputStream(), key_with_prefix)) {
      return false;
   }

   if (key_with_prefix.at(0) != '>') {
      throw FastaFormatException("Fasta key prefix '>' missing for key: " + key_with_prefix);
   }

   key = key_with_prefix.substr(1);
   return true;
}

bool silo::FastaReader::nextKey(std::string& key) {
   auto key_was_read = populateKey(key);

   in_file.getInputStream().ignore(LONG_MAX, '\n');
   return key_was_read;
}

bool silo::FastaReader::next(std::string& key, std::string& genome) {
   auto key_was_read = populateKey(key);
   if (!key_was_read) {
      return false;
   }

   if (!getline(in_file.getInputStream(), genome)) {
      throw FastaFormatException(
         "Missing genome sequence in line following key: " + key_with_prefix
      );
   }

   return true;
}
```
* Reading the genome or not is solved by delegation instead of by an additional parameter
* Why does it matter whether the key is mutated or not in case of an exception? If it's really such a big deal, then `populateKey` could return an optional string instead of populating the `key` reference.